packages/devtools_app/lib/src/code_size/code_size_screen.dart

```diff
@@ -274,34 +275,38 @@ class _SnapshotViewState extends State<SnapshotView> with AutoDisposeMixin {
 
   Widget _buildImportSnapshotView() {
     return ValueListenableBuilder(
-      valueListenable: controller.processingNotifier,
-      builder: (context, processing, _) {
-        if (processing) {
-          return Center(
-            child: Text(
-              CodeSizeScreen.loadingMessage,
-              textAlign: TextAlign.center,
-              style: TextStyle(
-                color: Theme.of(context).textTheme.headline1.color,
-              ),
-            ),
-          );
-        } else {
-          return Column(
-            children: [
-              Flexible(
-                child: FileImportContainer(
-                  title: 'Snapshot / APK analysis',
-                  instructions: SnapshotView.importInstructions,
-                  actionText: 'Analyze Snapshot / APK',
-                  onAction: controller.loadTreeFromJsonFile,
+        valueListenable: controller.processingNotifier,
+        builder: (context, processing, _) {
+          if (processing) {
+            return Center(
+              child: Text(
+                CodeSizeScreen.loadingMessage,
+                textAlign: TextAlign.center,
+                style: TextStyle(
+                  color: Theme.of(context).textTheme.headline1.color,
                 ),
               ),
-            ],
-          );
-        }
-      },
-    );
+            );
+          } else {
+            return Column(
+              children: [
+                Flexible(
+                  child: FileImportContainer(
+                    title: 'Snapshot / APK analysis',
+                    instructions: SnapshotView.importInstructions,
+                    actionText: 'Analyze Snapshot / APK',
+                    onAction: (jsonFile) {
+                      controller.loadTreeFromJsonFile(
+                        jsonFile,
+                        (error) => Notifications.of(context).push(error),
```

Comment:
we need to store Notifications.of(context) in a var before the callback here too

Reply:
Done